Calendar/Get-CalendarDiagnosticObjectsSummary.ps1

```diff
@@ -138,28 +139,30 @@ $ResponseTypeOptions = @{
 Run Get-CalendarDiagnosticObjects for passed in User with Subject or MeetingID.
 #>
 function GetCalendarDiagnosticObjects {
-
+    param(
+        [string]$Identity,
+        [string]$Subject,
+        [string]$MeetingID
+    )
     # Use MeetingID if we have it.
     if ($Identity -and $MeetingID) {
         Write-Verbose "Getting CalLogs for [$Identity] with MeetingID [$MeetingID]."
-        $script:InitialCDOs = Get-CalendarDiagnosticObjects -Identity $Identity -MeetingID $MeetingID -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults $LogLimit -ResultSize $LogLimit -ShouldBindToItem $true;
+        $CalLogs = Get-CalendarDiagnosticObjects -Identity $Identity -MeetingID $MeetingID -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults $LogLimit -ResultSize $LogLimit -ShouldBindToItem $true;
     }
 
     # Otherwise do a search on the subject.
-    if ($Identity -and $Subject -and !$MeetingID) {
+    if ($Identity -and $Subject) {
```

Comment:
When using a parameter set, it validates that both Subject and MeetingID are not passed in.
.\Get-CalendarDiagnosticObjectsSummary.ps1 Delegate -subject "Lunch" -MeetingID $MeetingID
_Get-CalendarDiagnosticObjectsSummary.ps1: Parameter set cannot be resolved using the specified named parameters. One or more parameters issued cannot be used together or an insufficient number of parameters were provided._

Reply:
Parameter sets is different than what I was talking about here. But you are able to use Parameter Sets as well. Because `Delegate` wasn't provided a parameter name, you might need to provide a position for the parameters to know what goes where.